(PNPM) When MTC encounters an external .d.marko with an absolute path outside of the rootDir, TS throws TS6059 and TS6307

### Version

3.1.2

### Description

When installing a dependency like ebayui-core via PNPM, PNPM creates a symlink:
```
node_modules/@ebay/ebayui-core
  → ../.pnpm/@ebay+ebayui-core@15.0.0_.../node_modules/@ebay/ebayui-core
```

If my project references `<ebay-chevron-right-16-icon/>`, MTC will throw: 

```
error TS6059: File '.../@ebay/ebayui-core/dist/components/ebay-icon/icons/ebay-chevron-right-16-icon/index.d.marko' is not under 'rootDir' '.../src'.
error TS6307: File '.../@ebay/ebayui-core/dist/components/ebay-icon/icons/ebay-chevron-right-16-icon/index.d.marko' is not listed within the file list of project '.../tsconfig.json'.
```

this does not happen with a component like ebay-button because the taglib can be found through `@ebay/ebayui-core`.  

```
@ebay/ebayui-core/marko.json
└── "tags-dir": "./dist/components"
    └── ebay-button/index.d.marko
```

ebay-icon- is nested in a taglib-imports. 
```
@ebay/ebayui-core/marko.json
└── "taglib-imports": ["./dist/components/ebay-icon/marko.json"]
    └── "tags-dir": "./icons"
        └── ebay-chevron-right-16-icon/index.d.marko
```

"tags-dir" scans a directory within the current taglib, so discovered tags inherit that taglib’s package metadata, including packageName: "@ebay/ebayui-core". "taglib-imports" resolves and loads another marko.json as a new taglib object using its absolute real path, but does not propagate the parent’s package identity. MTC consequently represents the imported tag using its absolute .pnpm path and does not classify its .d.marko file as an external declaration. In composite mode, TypeScript treats the file as project source and reports TS6059/TS6307 because it is outside rootDir and absent from the project’s input list.

### Reproduction

https://github.com/caseycarroll/pnpm-mtc-repro

--- a/.changeset/nested-taglib-imports-package-identity.md
+++ b/.changeset/nested-taglib-imports-package-identity.md
@@ -1,0 +1,5 @@
+---
+"@marko/compiler": patch
+---
+
+Propagate package identity through `taglib-imports` so tags from a nested imported taglib resolve by their package name instead of a realpath (eg pnpm's virtual store) that may not be importable.

--- a/packages/compiler/src/taglib/loader/Taglib.js
+++ b/packages/compiler/src/taglib/loader/Taglib.js
@@ -18,11 +18,33 @@ function handleImport(taglib, importedTaglib) {
 
   taglib.imports.push(importedTaglib);
 
+  // Tags from a taglib imported within the importing package inherit its identity,
+  // so they resolve by package name instead of a realpath (eg pnpm) that may not import.
+  if (
+    taglib.packageName &&
+    !importedTaglib.packageName &&
+    isWithin(taglib.packageRoot, importedTaglib.dirname)
+  ) {
+    importedTaglib.setPackageName(taglib.packageName, taglib.packageRoot);
+  }
+
   if (importedTaglib.imports) {
     importedTaglib.imports.forEach(function (nestedImportedTaglib) {
       handleImport(taglib, nestedImportedTaglib);
     });
   }
+}
+
+function isWithin(dir, filename) {
+  var rel = path.relative(dir, filename);
+  // `path.relative` walks out of the dir with `..`, or returns an absolute path
+  // when it cannot relate the two at all, eg across windows drives.
+  return (
+    !!rel &&
+    rel !== ".." &&
+    !rel.startsWith(`..${path.sep}`) &&
+    !path.isAbsolute(rel)
+  );
 }
 
 class Taglib {
@@ -32,6 +54,7 @@ class Taglib {
     this.isFromPackageJson = isFromPackageJson === true;
     this.dirname = path.dirname(this.filePath);
     this.packageName = packageName;
+    this.packageRoot = packageName ? this.dirname : undefined;
     this.scriptLang = undefined;
     this.tags = {};
     this.migrators = [];
@@ -76,8 +99,9 @@ class Taglib {
     this.setTagPackage(tag);
   }
 
-  setPackageName(packageName) {
+  setPackageName(packageName, packageRoot) {
     this.packageName = packageName;
+    this.packageRoot = packageRoot || this.dirname;
     for (var name in this.tags) {
       if (hasOwnProperty.call(this.tags, name)) {
         this.setTagPackage(this.tags[name]);
@@ -88,7 +112,7 @@ class Taglib {
   setTagPackage(tag) {
     if (this.packageName) {
       tag.packageName = this.packageName;
-      tag.packageRoot = this.dirname;
+      tag.packageRoot = this.packageRoot;
     }
   }
 

